src/script/page/RightSidebar/AddParticipants/AddParticipants.tsx

```diff
@@ -0,0 +1,334 @@
+    'isServicesRoom',
+    'isTeamOnly',
+    'participating_user_ids',
+  ]);
+  const {isTeam, teamMembers, teamUsers} = useKoSubscribableChildren(teamState, ['isTeam', 'teamMembers', 'teamUsers']);
+  const {connectedUsers, self: selfUser} = useKoSubscribableChildren(userState, ['connectedUsers', 'self']);
+  const {teamRole} = useKoSubscribableChildren(selfUser, ['teamRole']);
+  const {services} = useKoSubscribableChildren(integrationRepository, ['services']);
+
+  const {canManageServices} = generatePermissionHelpers(teamRole);
+
+  const [currentState, setCurrentState] = useState<PARTICIPANTS_STATE>(PARTICIPANTS_STATE.ADD_PEOPLE);
+  const isAddPeopleState = currentState === PARTICIPANTS_STATE.ADD_PEOPLE;
+  const isAddServiceState = currentState === PARTICIPANTS_STATE.ADD_SERVICE;
+
+  const [searchInput, setSearchInput] = useState<string>('');
+  const [contacts, setContacts] = useState<User[]>([]);
+  const [selectedContacts, setSelectedContacts] = useState<User[]>([]);
+
+  const [isInitialServiceSearch, setIsInitialServiceSearch] = useState<boolean>(true);
+
+  const enabledAddAction = selectedContacts.length > 0;
+
+  const headerText = selectedContacts.length
+    ? t('addParticipantsHeaderWithCounter', selectedContacts.length)
+    : t('addParticipantsHeader');
+
+  const showIntegrations = useMemo(() => {
+    const isServicesEnabled = isServicesRoom || isGuestAndServicesRoom;
+    const isService = !!firstUserEntity?.isService;
+    const allowIntegrations = isGroup || isService;
+
+    return isTeam && allowIntegrations && inTeam && !isTeamOnly && isServicesEnabled;
+  }, [isServicesRoom, isGuestAndServicesRoom, firstUserEntity, isGroup, isTeam, inTeam, isTeamOnly]);
+
+  const manageServicesUrl = getManageServicesUrl('client_landing');
+  const isSearching = searchInput.length > 0;
+
+  const onAddPeople = () => setCurrentState(PARTICIPANTS_STATE.ADD_PEOPLE);
+
+  const searchServices = async (value: string) => {
+    await integrationRepository.searchForServices(value);
+    setIsInitialServiceSearch(false);
+  };
+
+  const onAddServices = async () => {
+    setCurrentState(PARTICIPANTS_STATE.ADD_SERVICE);
+
+    await searchServices(searchInput);
+  };
+
+  const openManageServices = () => {
+    if (manageServicesUrl) {
+      safeWindowOpen(manageServicesUrl);
+    }
+  };
+
+  const onServiceSelect = (serviceEntity: ServiceEntity) => {
+    togglePanel(PanelViewModel.STATE.GROUP_PARTICIPANT_SERVICE, {
+      addMode: true,
+      entity: serviceEntity,
+    });
+  };
+
+  const addUsers = async () => {
+    const userEntities = selectedContacts.slice();
+
+    await conversationRepository.addUsers(activeConversation, userEntities);
+  };
+
+  const onAddParticipants = async () => {
+    await addUsers();
+    onBack();
+  };
+
+  const onSearchInput = async (value: string) => {
+    await searchServices(value);
+    setSearchInput(value);
+  };
+
+  const getUserEntities = useCallback(() => {
```

Comment:
Is this function necessary here? We could just use `useMemo`, so there is no need to call this function on initial render (line 187) - we lose 1 rerender?